src/sources/redis/mod.rs

```diff
@@ -216,7 +216,7 @@ impl SourceConfig for RedisSourceConfig {
                 Self::NAME,
                 redis_key_path,
                 &owned_value_path!("key"),
-                Kind::bytes(),
+                Kind::bytes().or_undefined(),
```

Comment:
Looking [here](https://github.com/vectordotdev/vector/blob/41b0a276c13594d2853613c64138b4f73e4df237/src/sources/redis/mod.rs#L277) it looks like there's opportunity of not having that value at runtime?

Reply:
:thinking: Was going to comment similarly- I think the doc string for the redis key config [here](https://github.com/vectordotdev/vector/blob/41b0a276c13594d2853613c64138b4f73e4df237/src/sources/redis/mod.rs#L119) , is misleading ?
It looks to me like the event will always contain that field and it's just that we are not overwriting it with if it already exists.